validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/SetFeeRecipient.java

```diff
@@ -0,0 +1,126 @@
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiEndpoint;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.datastructures.eth1.Eth1Address;
+import tech.pegasys.teku.validator.client.BeaconProposerPreparer;
+import tech.pegasys.teku.validator.client.SetFeeRecipientException;
+
+public class SetFeeRecipient extends RestApiEndpoint {
+  public static final String ROUTE = "/eth/v1/validator/{pubkey}/feerecipient";
+
+  private static final DeserializableTypeDefinition<SetFeeRecipientBody>
+      FEE_RECIPIENT_REQUEST_BODY =
+          DeserializableTypeDefinition.object(SetFeeRecipientBody.class)
+              .name("SetFeeRecipientBody")
+              .initializer(SetFeeRecipientBody::new)
+              .withField(
+                  "ethaddress",
+                  ETH1ADDRESS_TYPE,
+                  SetFeeRecipientBody::getEth1Address,
+                  SetFeeRecipientBody::setEth1Address)
+              .build();
+
+  private final Optional<BeaconProposerPreparer> beaconProposerPreparer;
+
+  public SetFeeRecipient(final Optional<BeaconProposerPreparer> beaconProposerPreparer) {
+    super(
+        EndpointMetadata.post(ROUTE)
+            .operationId("SetFeeRecipient")
+            .withBearerAuthSecurity()
+            .pathParam(PARAM_PUBKEY_TYPE)
+            .summary("Set validator fee recipient")
+            .requestBodyType(FEE_RECIPIENT_REQUEST_BODY)
+            .description(
+                "Sets the validator client fee recipient mapping which will then update the beacon node. "
+                    + "Existing mappings for the same validator public key will be overwritten. "
+                    + "Configuration file settings will take precedence over this API, so if your validator fee recipient "
+                    + "configuration file contains this public key, it will need to be removed before attempting to update with this api."
+                    + "Cannot specify default fee recipient address through API.\n\n"
+                    + "WARNING: The fee_recipient is not used on Phase0 or Altair networks.")
+            .response(SC_ACCEPTED, "Success")
+            .response(SC_SERVICE_UNAVAILABLE, "Unable to update fee recipient at this time")
+            .withAuthenticationResponses()
+            .withNotFoundResponse()
+            .build());
+    this.beaconProposerPreparer = beaconProposerPreparer;
+  }
+
+  @Override
+  public void handleRequest(final RestApiRequest request) throws JsonProcessingException {
+    final BLSPublicKey publicKey = request.getPathParameter(PARAM_PUBKEY_TYPE);
+    final SetFeeRecipientBody body = request.getRequestBody();
+    if (beaconProposerPreparer.isEmpty()) {
+      request.respondError(
+          SC_SERVICE_UNAVAILABLE, "Not able to update fee recipient at this time.");
```

Comment:
Is this just because Bellatrix isn't enabled? Might be worth being explicit about why it's not available.